tests/unit/test_endpoint_provider.py

```diff
@@ -87,13 +87,21 @@ def __init__(self, *args, **kwargs):
         )
 
 
+@patch("charms.observability_libs.v0.juju_topology.JujuTopology.is_valid_uuid", lambda *args: True)
```

Comment:
Is the class level patch needed in addition to the method level patches?

Reply:
Yes, in all cases. Patches to `setUp` do not apply to methods if it's instantiated later, and class-level patches do not apply to `setUp`